link/xdp.go

```diff
@@ -0,0 +1,72 @@
+package link
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/internal/sys"
+)
+
+// XDPAttachTypeFlags represents the XDP attach type.
+type XDPAttachTypeFlags uint32
```

Comment:
The name is a bit cumbersome. `XDPMode`? `XDPAttachFlags`? `XDPAttachType`?

Reply:
Updated to XDPAttachFlags.